src/components/Sidebar/PostRecommendation.js

```diff
@@ -77,8 +75,8 @@ class PostRecommendation extends Component {
     }
   };
 
-  navigateToPostComments = (category, author, permlink) => {
-    this.props.history.push(`/${category}/@${author}/${permlink}#comments`);
+  navigateToPostComments = () => {
+    document.getElementById('comments').scrollIntoView();
     this.setState(this.state);
```

Comment:
What's the purpose of `this.setState(this.state)`? If it's to force re-render we should use `forceUpdate()` instead. Is there no other way around this than to force re-render? I have problems understanding what is going on in this component to be honest, but as far as I understood it I think we should re-render only when props or state changes, so we should implement this logic in `shouldComponentUpdate`.
This code got to our code base earlier and is not really related to this PR, so I won't use this PR for solving this problem, but I think we should improve logic there in another PR.

Reply:
It's to force an update. I read online that it was bad practice to use forceUpdate.

Reply:
Yes it is, but `this.setState(this.state)` is equally as bad or even worse as it would trigger more lifecycle methods. Bad practise is better than bad practise + hacking it around.

Reply:
For some reason I thought that forceUpdate would be slower then this.state, since it forces a re-render of all child components, but looking in deeper and testing it out, I think forceUpdate might faster, since it doesn't go through shouldComponentUpdate, will change.. thanks for the feedback